inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp

```diff
@@ -79,174 +87,347 @@ void MKLDNNPadNode::initSupportedPrimitiveDescriptors() {
+        size_t dstIdx = 0;
+        getDstIdx(indexes, dstIdx);
+
+        for (size_t iwork = start; iwork < end; ++iwork, dstIdx += params.lastDstDim) {
+            size_t srcIdx = 0;
+            for (size_t i = 0; i < params.nDimsForWork; ++i) {
+                size_t idx = (indexes[i] < padsBegin[i]) ? (padsBegin[i] - indexes[i] - shift) :
+                             ((indexes[i] >= params.srcODims[i]) ? (params.srcDimsForReflectOrSymmetric[i] - indexes[i]) : (indexes[i] - padsBegin[i]));
                 srcIdx += idx * params.srcStrides[i];
             }
+            srcIdx *= params.sizeData;
 
-            for (size_t i = 0; i < padsBegin[dimsSize_1]; ++i)
-                dstData[dstIdx + i] = srcData[srcIdx + padsBegin[dimsSize_1] - i - shift];
+            for (size_t i = 0; i < padsBegin[params.nDimsForWork]; ++i)
+                cpu_memcpy(&dstData[dstIdx + i * params.shift],
+                           &srcData[srcIdx + (padsBegin[params.nDimsForWork] - shift - i) * params.shift], params.shift);
 
-            cpu_memcpy(&dstData[dstIdx + padsBegin[dimsSize_1]], &srcData[srcIdx],
-                       sizeof(float) * srcDims[dimsSize_1]);
+            cpu_memcpy(&dstData[dstIdx + padsBegin[params.nDimsForWork] * params.shift], &srcData[srcIdx],
+                       params.srcDims[params.nDimsForWork] * params.shift);
 
-            for (size_t i = params.srcODims[dimsSize_1]; i < dstDims[dimsSize_1]; ++i)
-                dstData[dstIdx + i] = srcData[srcIdx + src_2[dimsSize_1] - i];
+            size_t srcShift = (params.srcDimsForReflectOrSymmetric[params.nDimsForWork] - params.srcODims[params.nDimsForWork]) * params.shift;
+            for (size_t i = 0; i < padsEnd[params.nDimsForWork]; ++i)
+                cpu_memcpy(&dstData[dstIdx + (params.srcODims[params.nDimsForWork] + i) * params.shift],
+                           &srcData[srcIdx + srcShift - i * params.shift], params.shift);
 
-            parallel_step(dimsSize_1, counters, dstDims);
+            parallel_step(params.nDimsForWork, params.dstDims, indexes);
         }
     });
 }
 
-size_t MKLDNNPadNode::getWorkAmountDst() const {
-    auto dstDims = getChildEdgeAt(0)->getDims().ToSizeVector();
-    return params.dstStrides[0] * dstDims[0] / dstDims[dstDims.size() - 1];
+inline void MKLDNNPadNode::getDstIdx(const InferenceEngine::SizeVector& indexes, size_t& dstIdx) const {
+    for (size_t i = 0; i < params.nDimsForWork; ++i)
+        dstIdx += indexes[i] * params.dstStrides[i];
+    dstIdx *= (padMode == CONSTANT && padValue != 0) ? 1 : params.sizeData;
+}
+
+inline uint8_t* MKLDNNPadNode::getDataPtr(const MKLDNNMemory& memoryPtr) const {
+    return reinterpret_cast<uint8_t*>(memoryPtr.GetData()) + memoryPtr.GetDescriptor().data.layout_desc.blocking.offset_padding *
+                                      MKLDNNExtensionUtils::sizeOfDataType(mkldnn::memory::data_type(memoryPtr.GetDescriptor().data.data_type));
```

Comment:
The offset_padding has an ptrdiff_t type, so it seems to me that there is no need to multiply this value by the element size since this is an already calculated offset to the beginning address. One can check MKLDNNMemory::SetData method to confirm my statement.

Reply:
You're right. Thank you!

Reply:
Sorry, I was wrong. Since we calculate offset with int8 ptrs, we have to multiply the offset by the data size, because ptr arithmetic implies address calculation in data blocks with the size of data type, and actually the offset_padding is in the data blocks but not in bytes.

Reply:
Ok, I returned back